test/unittests/t_mountpoint.cc

```diff
@@ -843,3 +844,57 @@ TEST_F(T_MountPoint, History) {
     EXPECT_EQ(loader::kFailOk, mp->boot_status());
   }
 }
+
+
+TEST_F(T_MountPoint, MaxServers) {
+  CreateMiniRepository();
+  string server_url;
+  ASSERT_TRUE(options_mgr_.GetValue("CVMFS_SERVER_URL", &server_url));
+  server_url += ";" + server_url;
+  options_mgr_.SetValue("CVMFS_SERVER_URL", server_url);
+
+  UniquePtr<FileSystem> fs(FileSystem::Create(fs_info_));
+  ASSERT_EQ(loader::kFailOk, fs->boot_status());
+  std::vector<std::string> host_chain;
+
+  options_mgr_.SetValue("CVMFS_MAX_SERVERS", "10");
+  {
+    UniquePtr<MountPoint> mp(MountPoint::Create("keys.cern.ch", fs.weak_ref()));
+    ASSERT_EQ(loader::kFailOk, mp->boot_status());
+    mp->download_mgr()->GetHostInfo(&host_chain, NULL, NULL);
+    EXPECT_EQ(2U, host_chain.size());
+    mp->external_download_mgr()->GetHostInfo(&host_chain, NULL, NULL);
+    EXPECT_EQ(1U, host_chain.size());
+  }
+
+  options_mgr_.SetValue("CVMFS_EXTERNAL_MAX_SERVERS", "10");
+  {
+    UniquePtr<MountPoint> mp(MountPoint::Create("keys.cern.ch", fs.weak_ref()));
+    ASSERT_EQ(loader::kFailOk, mp->boot_status());
+    mp->external_download_mgr()->GetHostInfo(&host_chain, NULL, NULL);
+    EXPECT_EQ(1U, host_chain.size());
```

Comment:
Why is this value 1 when it is 2 below where CVMFS_EXTERNAL_MAX_SERVERS=0?

Reply:
I added a comment, as for the settings below: we explicitly set `CVMFS_EXTERNAL_URL` only further down (`options_mgr_.SetValue("CVMFS_EXTERNAL_URL", server_url);`). The `CVMFS_EXTERNAL_MAX_SERVERS=0` setting means "no limit".